graphql.go

```diff
@@ -231,7 +236,15 @@ func (c *Client) doRaw(ctx context.Context, op operationType, v interface{}, var
 // do executes a single GraphQL operation and unmarshal json.
 func (c *Client) do(ctx context.Context, op operationType, v interface{}, variables map[string]interface{}, options ...Option) error {
 	data, resp, respBuf, errs := c.buildAndRequest(ctx, op, v, variables, options...)
+	return c.processResponse(v, data, resp, respBuf, errs)
+}
+
+func (c *Client) DoWithQuery(ctx context.Context, query string, v interface{}, variables map[string]interface{}, options ...Option) error {
```

Comment:
```suggestion
func (c *Client) Do(ctx context.Context, query string, v interface{}, variables map[string]interface{}, options ...Option) error {
```

or we can name it `Exec`

Reply:
Thanks, `Exec` sounds certainly better. I renamed it and added a docstring. Thanks!